src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -184,11 +186,21 @@ public FileUploadSubmission save(FileUploadSubmission fileUploadSubmission, Mult
         if (exerciseDueDate != null && exerciseDueDate.isBefore(ZonedDateTime.now()) && participation.getInitializationDate().isBefore(exerciseDueDate)) {
             throw new ResponseStatusException(HttpStatus.FORBIDDEN);
         }
+        if (file.isEmpty()) {
+            throw new IOException("Failed to store empty file" + file.getName());
+        }
+        var multipartFileHash = DigestUtils.md5Hex(file.getInputStream());
         // check if we already had file associated with this submission
         fileUploadSubmission.onDelete();
 
         final var localPath = saveFileForSubmission(file, fileUploadSubmission, exercise);
 
+        // We need to ensure that we can access the store file and the stored file is the same as was passed to us in the request
+        var storedFileHash = DigestUtils.md5Hex(Files.newInputStream(Path.of(localPath)));
+        if (!multipartFileHash.equals(storedFileHash)) {
+            throw new IOException("The file " + file.getName() + "could not be stored");
+        }
+
```

Comment:
Since I could not highlight the Transactional annotation above, I'm writing my comment here :) 
Do we need `@Transactional(rollbackFor = Exception.class)` for the save method? If not, then please remove it.

Reply:
This change is a part of a PR https://github.com/ls1intum/Artemis/pull/1107, so I didn't do it here yet. But now it is done